README.md

```diff
@@ -24,6 +24,68 @@ npm run start:debug
 
 - And run .vscode/launch.json (Debug Attach)
 
+# Adding CORS support
+
+Cross-Origin Resource Sharing (CORS) is a W3C specification and mechanism that you can use to request restricted resources from a domain outside the current domain. In other words, CORS is a technique for consuming an API served from an origin different than yours ([more information](https://www.freecodecamp.org/news/the-terrible-performance-cost-of-cors-api-on-the-single-page-application-spa-6fcf71e50147/)).
+
+If your api is going to be consumed by client code running web applications under different domains, you may need to enable
+CORS access on your express server.
+
+[More information about CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+
+How to configure CORS in express:
+
+First install the following package:
+
+```bash
+npm install cors -P
+```
+
+Thenn install the typings for this package:
+
+```bash
+npm install @types/cors -D
+```
+
+_/src/app.ts_
+
+```diff
+import { createApp } from './express.server';
++ import cors from 'cors';
+import { envConstants } from './env.constants';
+import { api } from './api';
+
+const app = createApp();
+
++ // Warning: narrow down your CORS allowed origins to the web application domains that
++ // are allowed to access this application.
++ // check the methods and origin properties in the CorsOptions
++ const options: cors.CorsOptions = {
++  allowedHeaders: [
++    'Origin',
++    'X-Requested-With',
++    'Content-Type',
++    'Accept',
++    'X-Access-Token',
++  ],
++  credentials: true,
++  // IMPORTANT YOU MAY LIMIT THE ALLOWED VERBS TO FOR INSTANCE ONLY GET
++  methods: 'GET,HEAD,OPTIONS,PUT,PATCH,POST,DELETE',
++  // IMPORTANT LIMIT THIS HERE TO YOUR CLIENT APPS DOMAINS
++  origin: '*',
++  preflightContinue: false,
```

Comment:
Is this property required?

Reply:
Not sure, something we have to check

Reply:
Is added by default in configuration in the cors repo. here is explained: https://github.com/expressjs/cors/issues/153
if I understood right, is too keep the cors in the middleware when you jump with next() method.